File: examples/TanksMiniGameProject/assets/unity/WXScripts.mgepackage/src/ShellPool.js
Bridge.assembly("minigame-adaptor-project", function ($asm, globals) {
    "use strict";

    Bridge.define("ShellPool", {
        statics: {
            fields: {
                m_freeShellList: null
            },
            ctors: {
                init: function () {
                    this.m_freeShellList = new (System.Collections.Generic.List$1(MiniGameAdaptor.GameObject)).ctor();
                }
            },
            methods: {
                Create: function (original, position, rotation) {
                    if (ShellPool.m_freeShellList.Count === 0) {
                        var gameObject = MiniGameAdaptor.Object.Instantiate$3(MiniGameAdaptor.GameObject, original, position.$clone(), rotation.$clone());
                        gameObject.SetActive(true);
                        return gameObject;
                    } else {
                        var gameObject1 = ShellPool.m_freeShellList.getItem(((ShellPool.m_freeShellList.Count - 1) | 0));
                        ShellPool.m_freeShellList.removeAt(((ShellPool.m_freeShellList.Count - 1) | 0));
                        gameObject1.SetActive(true);
                        gameObject1.transform.position = position.$clone();
                        gameObject1.transform.rotation = rotation.$clone();
                        return gameObject1;
                    }
                },
                Destroy: function (gameObject) {
                    gameObject.SetActive(false);
                    ShellPool.m_freeShellList.add(gameObject);
                }
            }
        }
    });
});

File: examples/TanksMiniGame/Assets/_Completed-Assets/Scripts/Shell/ShellPool.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
public class ShellPool
{
  static List<GameObject> m_freeShellList = new List<GameObject>();
  public static GameObject Create(GameObject original, Vector3 position, Quaternion rotation)
  {
    if (m_freeShellList.Count == 0)
    {
      GameObject gameObject = UnityEngine.GameObject.Instantiate(original, position, rotation) as GameObject;
      gameObject.SetActive(true);
      return gameObject;
    }
    else
    {
      GameObject gameObject = m_freeShellList[m_freeShellList.Count - 1];
      m_freeShellList.RemoveAt(m_freeShellList.Count - 1);
      gameObject.SetActive(true);
      gameObject.transform.position = position;
      gameObject.transform.rotation = rotation;
      return gameObject;
    }
  }
  public static void Destroy(GameObject gameObject)
  {
    gameObject.SetActive(false);
    m_freeShellList.Add(gameObject);
  }

}